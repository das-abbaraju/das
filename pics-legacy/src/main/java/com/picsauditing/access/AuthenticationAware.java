package com.picsauditing.access;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Retention(RetentionPolicy.RUNTIME)
public @interface AuthenticationAware {
}
