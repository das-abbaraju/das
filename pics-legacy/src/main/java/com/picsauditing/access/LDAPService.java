package com.picsauditing.access;

import com.picsauditing.featuretoggle.Features;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.AuthenticationException;
import org.springframework.security.ldap.authentication.ad.ActiveDirectoryLdapAuthenticationProvider;

import javax.security.auth.login.FailedLoginException;


public class LDAPService {

    public static final String PICS_CORP = "@PICS.CORP";

    private static final Logger logger = LoggerFactory.getLogger(LDAPService.class);

    public static final String PICSAD = "picsad";

    @Autowired
    private ActiveDirectoryLdapAuthenticationProvider ldapActiveDirectoryAuthProvider;

    public boolean doLDAPLoginAuthentication(String identityProvider, String username, String password) throws FailedLoginException {
        if (Features.USE_LDAP_AUTHENTICATION.isActive()) {
            switch (identityProvider) {
                case PICSAD:
                    return doPICSLdapAuthentication(username, password);
            }
        }
        return false;
    }

    private String getPICSLdapUser(String username) {
        StringBuilder ldapUser = new StringBuilder();
        ldapUser.append(username);
        ldapUser.append(PICS_CORP);
        return ldapUser.toString();
    }

    private boolean doPICSLdapAuthentication(String username, String password) {
        ldapActiveDirectoryAuthProvider.setConvertSubErrorCodesToExceptions(true);
        String ldapUser = getPICSLdapUser(username);
        try {
            UsernamePasswordAuthenticationToken authentication = new UsernamePasswordAuthenticationToken(ldapUser, password);
            Authentication result = ldapActiveDirectoryAuthProvider.authenticate(authentication);
            return result != null ? result.isAuthenticated() : false;
        } catch (AuthenticationException ace) {
            logger.error("Bad LDAP Credentials for user: " + username);
        }
        return false;
    }
}
