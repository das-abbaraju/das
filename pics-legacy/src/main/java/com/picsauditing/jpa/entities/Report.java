package com.picsauditing.jpa.entities;

import com.picsauditing.report.fields.FieldType;
import com.picsauditing.report.fields.ReportField;
import com.picsauditing.report.models.ModelType;
import com.picsauditing.report.tables.FieldImportance;
import org.apache.commons.collections.CollectionUtils;
import org.hibernate.annotations.SQLDelete;
import org.hibernate.annotations.Where;

import javax.persistence.*;
import javax.persistence.Column;
import java.util.ArrayList;
import java.util.Collections;
import java.util.Comparator;
import java.util.List;

@SuppressWarnings("serial")
@Entity
@Table(name = "report")
@SQLDelete(sql = "UPDATE report SET deleted = 1 WHERE id = ?")
@Where(clause = "deleted = 0")
public class Report extends BaseTable {

	private ModelType modelType;
	private User owner;
	private String name;
	private String description;
	private String filterExpression;
	private String sql;
	private boolean deleted;
	private boolean isPublic;
    private boolean groupBy;

	private List<com.picsauditing.jpa.entities.Column> columns = new ArrayList<com.picsauditing.jpa.entities.Column>();
	private List<Filter> filters = new ArrayList<Filter>();
	private List<Sort> sorts = new ArrayList<Sort>();

	private List<ReportPermissionUser> reportPermissionUsers = new ArrayList<ReportPermissionUser>();
    private List<ReportPermissionAccount> reportPermissionAccounts = new ArrayList<ReportPermissionAccount>();
	private List<ReportUser> reportUsers = new ArrayList<ReportUser>();

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "ownerID", nullable = false)
	public User getOwner() {
		return owner;
	}

	public void setOwner(User owner) {
		this.owner = owner;
	}

	@Enumerated(EnumType.STRING)
	@Column(nullable = false)
	@ReportField(importance = FieldImportance.Required, width = 200)
	public ModelType getModelType() {
		return modelType;
	}

	public void setModelType(ModelType type) {
		this.modelType = type;
	}

	@Column(nullable = false)
	@ReportField(importance = FieldImportance.Required, width = 200)
	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

    @ReportField
	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	@Transient
	public String getSql() {
		return sql;
	}

	public void setSql(String sql) {
		this.sql = sql;
	}

	public boolean isDeleted() {
		return deleted;
	}

	public void setDeleted(boolean deleted) {
		this.deleted = deleted;
	}

    @ReportField
    public String getFilterExpression() {
        return filterExpression;
    }

    public void setFilterExpression(String filterExpression) {
        this.filterExpression = filterExpression;
    }

    @OneToMany(mappedBy = "report", cascade = CascadeType.ALL, orphanRemoval = true)
	public List<com.picsauditing.jpa.entities.Column> getColumns() {
		return columns;
	}

	public void setColumns(List<com.picsauditing.jpa.entities.Column> columns) {
		this.columns = columns;
	}

	public void addColumn(com.picsauditing.jpa.entities.Column column) {
		column.setReport(this);
		columns.add(column);
	}

	@OneToMany(mappedBy = "report", cascade = CascadeType.ALL, orphanRemoval = true)
	public List<Filter> getFilters() {
		return filters;
	}

	public void setFilters(List<Filter> filters) {
		this.filters = filters;
	}

	public void addFilter(Filter filter) {
		filter.setReport(this);
		filters.add(filter);
	}

	@OneToMany(mappedBy = "report", cascade = CascadeType.ALL, orphanRemoval = true)
	public List<Sort> getSorts() {
		return sorts;
	}

	public void setSorts(List<Sort> sorts) {
		this.sorts = sorts;
	}

	public void addSort(Sort sort) {
		sort.setReport(this);
		sorts.add(sort);
	}

	@Override
	public String toString() {
		return name;
	}

	@OneToMany(mappedBy = "report", cascade = CascadeType.ALL)
	public List<ReportUser> getReportUsers() {
		return reportUsers;
	}

	public void setReportUsers(List<ReportUser> reportUsers) {
		this.reportUsers = reportUsers;
	}

	public boolean hasNoColumns() {
		return CollectionUtils.isEmpty(columns);
	}

	public boolean hasNoModelType() {
		return modelType == null;
	}

	public boolean hasNoOwner() {
		return owner == null || owner.getId() == 0;
	}

	@OneToMany(mappedBy = "report", cascade = CascadeType.ALL)
	public List<ReportPermissionUser> getReportPermissionUsers() {
		return reportPermissionUsers;
	}

	public void setReportPermissionUsers(List<ReportPermissionUser> reportPermissionUsers) {
		this.reportPermissionUsers = reportPermissionUsers;
	}

    @OneToMany(mappedBy = "report", cascade = CascadeType.ALL)
    public List<ReportPermissionAccount> getReportPermissionAccounts() {
        return reportPermissionAccounts;
    }

    public void setReportPermissionAccounts(List<ReportPermissionAccount> reportPermissionAccounts) {
        this.reportPermissionAccounts = reportPermissionAccounts;
    }

    public void sortColumns() {
		Collections.sort(columns, new Comparator<com.picsauditing.jpa.entities.Column>() {
			@Override
			public int compare(com.picsauditing.jpa.entities.Column c1, com.picsauditing.jpa.entities.Column c2) {
				if (c1.getSortIndex() == c2.getSortIndex()) {
					return c1.getId() - c2.getId();
				}

				return c1.getSortIndex() - c2.getSortIndex();
			}
		});
	}

    @ReportField(type = FieldType.Boolean)
    public boolean isPublic() {
		return isPublic;
	}

	public void setPublic(boolean isPublic) {
		this.isPublic = isPublic;
	}

    @Transient
    public boolean hasGroupBy() {
        return groupBy;
    }

    public void setGroupBy(boolean groupBy) {
        this.groupBy = groupBy;
    }
}