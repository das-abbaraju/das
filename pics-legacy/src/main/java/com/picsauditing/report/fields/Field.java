package com.picsauditing.report.fields;

import java.util.HashSet;
import java.util.Set;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

import javax.persistence.Transient;

import com.picsauditing.access.OpPerms;
import com.picsauditing.access.Permissions;
import com.picsauditing.dr.domain.fields.DisplayType;
import com.picsauditing.dr.domain.fields.FilterType;
import com.picsauditing.jpa.entities.Column;
import com.picsauditing.report.tables.FieldImportance;
import com.picsauditing.util.Strings;

public class Field {

	private static final Pattern FIELD_VARIABLE_PATTERN = Pattern.compile("\\{(\\w+)\\}");

	private FieldType type = FieldType.String;
	private String name;
	private String categoryTranslation;
	private String text;
	private String url;
	private int width = Column.DEFAULT_WIDTH;
	private String help;

    private String requiredJoin;
	private Class<?> fieldClass;
	private String databaseColumnName;
	private boolean visible = true;
	private boolean filterable = true;
	private boolean sortable = true;
	private String preTranslation;
	private String postTranslation;
	private String separator;
	private String drillDownField;
	private OpPerms requiredPermission = OpPerms.None;
	private FieldImportance importance = FieldImportance.Low;

    private String prefixValue;
    private String suffixValue;

	public Field(ReportField annotation) {
		type = annotation.type();
		width = annotation.width();

		if (Strings.isEmpty(annotation.url())) {
			url = null;
		} else {
			url = annotation.url();
		}

		requiredPermission = annotation.requiredPermissions();
		visible = annotation.visible();
		filterable = annotation.filterable();
		sortable = annotation.sortable();
		importance = annotation.importance();

		preTranslation = annotation.i18nKeyPrefix();
		postTranslation = annotation.i18nKeySuffix();
		if (type.getFilterType() == FilterType.Multiselect && Strings.isEmpty(preTranslation)) {
			preTranslation = type.toString();
		}
	}

	public Field(String name) {
		this.name = name;
		this.databaseColumnName = name;
	}

	public Field(String name, String databaseColumnName, FieldType type) {
		this.name = name;
		this.databaseColumnName = databaseColumnName;
		if (type == null) {
			throw new RuntimeException("type is required when creating Fields");
		}

		this.type = type;
	}

	public Field setTranslationPrefixAndSuffix(String prefix, String suffix) {
		this.preTranslation = prefix;
		this.postTranslation = suffix;
		return this;
	}

	public String getI18nKey(String value) {
		String key = value;

		if (!Strings.isEmpty(preTranslation)) {
			key = preTranslation + "." + key;
		}

		if (!Strings.isEmpty(postTranslation)) {
			key = key + "." + postTranslation;
		}

		return key;
	}

	public Set<String> getDependentFields() {
		Set<String> dependent = new HashSet<String>();
		if (Strings.isNotEmpty(url)) {
			Matcher urlFieldMatcher = FIELD_VARIABLE_PATTERN.matcher(url);

			while (urlFieldMatcher.find()) {
				dependent.add(urlFieldMatcher.group(1));
			}
		}
        if (drillDownField != null)
            dependent.add(drillDownField);

		return dependent;
	}

	public boolean isTranslated() {
		if (Strings.isEmpty(preTranslation) && Strings.isEmpty(postTranslation)) {
			return false;
		}

		return true;
	}

	public FieldType getType() {
		return type;
	}

	public void setType(FieldType type) {
		this.type = type;
	}

	@Transient
	public DisplayType getDisplayType() {
		return type.getDisplayType();
	}

	@Transient
	public FilterType getFilterType() {
		return type.getFilterType();
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getText() {
		return text;
	}

	public void setText(String text) {
		this.text = text;
	}

	public String getDatabaseColumnName() {
		return databaseColumnName;
	}

	public void setDatabaseColumnName(String databaseColumnName) {
		this.databaseColumnName = databaseColumnName;
	}

	public int getWidth() {
		return width;
	}

	public void setWidth(int width) {
		this.width = width;
	}

	public String getHelp() {
		return help;
	}

	public void setHelp(String help) {
		this.help = help;
	}

	public String getUrl() {
		return url;
	}

	public void setUrl(String url) {
		this.url = url;
	}

	@Transient
	public String getCategoryTranslation() {
		return categoryTranslation;
	}

	public void setCategoryTranslation(String categoryTranslation) {
		this.categoryTranslation = categoryTranslation;
	}

	public void setPreTranslation(String preTranslation) {
		this.preTranslation = preTranslation;
	}

	public void setPostTranslation(String postTranslation) {
		this.postTranslation = postTranslation;
	}

    public String getSeparator() {
        return separator;
    }

    public void setSeparator(String separator) {
        this.separator = separator;
    }

	public Field requirePermission(OpPerms opPerm) {
		requiredPermission = opPerm;
		return this;
	}

	public boolean canUserSeeQueryField(Permissions permissions) {
		if (requiredPermission == null) {
			return true;
		}

		if (requiredPermission.isNone()) {
			return true;
		}

		return permissions.hasPermission(requiredPermission);
	}

	public void setFieldClass(Class<?> fieldClass) {
		this.fieldClass = fieldClass;
	}

	public Class<?> getFieldClass() {
		return fieldClass;
	}

	public String getPreTranslation() {
		return preTranslation;
	}

	public String getPostTranslation() {
		return postTranslation;
	}

	// TODO: This is very questionable
	public boolean isVisible() {
		return visible;
	}

	public void setVisible(boolean visible) {
		this.visible = visible;
	}

	public boolean isFilterable() {
		return filterable;
	}

	public void setFilterable(boolean filterable) {
		this.filterable = filterable;
	}

	public boolean isSortable() {
		return sortable;
	}

	public void setSortable(boolean sortable) {
		this.sortable = sortable;
	}

	public FieldImportance getImportance() {
		return importance;
	}

	public void setImportance(FieldImportance importance) {
		this.importance = importance;
	}

    public String getPrefixValue() {
        return prefixValue;
    }

    public void setPrefixValue(String prefixValue) {
        this.prefixValue = prefixValue;
    }

    public String getSuffixValue() {
        return suffixValue;
    }

    public void setSuffixValue(String suffixValue) {
        this.suffixValue = suffixValue;
    }

    public String getDrillDownField() {
        return drillDownField;
    }

    public void setDrillDownField(String drillDownField) {
        this.drillDownField = drillDownField;
    }

    public Field clone() {
		Field copiedField = new Field(name, databaseColumnName, type);
		copiedField.text = text;
		copiedField.url = url;
		copiedField.width = width;
		copiedField.help = help;
		copiedField.fieldClass = fieldClass;
		copiedField.visible = visible;
		copiedField.filterable = filterable;
		copiedField.sortable = sortable;
        copiedField.requiredJoin = requiredJoin;
		copiedField.preTranslation = preTranslation;
		copiedField.postTranslation = postTranslation;
        copiedField.separator = separator;
        copiedField.drillDownField = drillDownField;
		copiedField.requiredPermission = requiredPermission;
		copiedField.importance = importance;
        copiedField.prefixValue = prefixValue;
        copiedField.suffixValue = suffixValue;

		return copiedField;
	}

	@Override
	public String toString() {
		return categoryTranslation + ": " + name;
	}

    public String getRequiredJoin() {
        return requiredJoin;
    }

    public void setRequiredJoin(String requiredJoin) {
        this.requiredJoin = requiredJoin;
    }
}