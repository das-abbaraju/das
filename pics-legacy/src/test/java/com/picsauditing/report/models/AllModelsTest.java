package com.picsauditing.report.models;

import com.google.common.base.Joiner;
import com.picsauditing.EntityFactory;
import com.picsauditing.access.OpPerms;
import com.picsauditing.access.Permissions;
import com.picsauditing.dr.domain.fields.QueryFilterOperator;
import com.picsauditing.jpa.entities.Filter;
import com.picsauditing.jpa.entities.Report;
import com.picsauditing.report.SqlBuilder;
import com.picsauditing.report.fields.Field;
import com.picsauditing.search.SelectSQL;
import org.approvaltests.Approvals;
import org.approvaltests.reporters.DiffReporter;
import org.approvaltests.reporters.UseReporter;
import org.junit.Before;
import org.junit.Ignore;
import org.junit.Test;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Map;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertTrue;

@UseReporter(DiffReporter.class)
public class AllModelsTest {

    private AbstractModel model;
    private Permissions permissions;

    @Before
    public void setUp() {
        permissions = EntityFactory.makePermission();
    }

    @Test
    public void testAllModels() throws Exception {
        StringBuilder actual = new StringBuilder();

        for (ModelType type : ModelType.values()) {
            actual.append(String.format("Model: %s\nPermission: %s\n----------------------------------------\n", type, permissions));
            permissions = EntityFactory.makePermission();
            model = ReportModelFactory.build(type, permissions);
            if (model != null) {
                actual.append(getJoin());
            } else {
                actual.append("Not Yet Implemented");
            }
            actual.append("\n\n");
        }
        Approvals.verify(actual.toString());
    }

    @Test
    @Ignore
    public void testAllSql() throws Exception {
        SqlBuilder builder = new SqlBuilder();
        StringBuilder actual = new StringBuilder();

        for (ModelType type : ModelType.values()) {
            actual.append(String.format("Model: %s\nPermission: %s\n----------------------------------------\n", type, permissions));
            permissions = EntityFactory.makePermission();
            try {
                final Report report = new Report();
                model = ReportModelFactory.build(type, permissions);
                report.setModelType(type);
                final SelectSQL selectSQL = builder.initializeReportAndBuildSql(report, permissions);
                actual.append(selectSQL.toString());
            } catch (Exception e) {
                actual.append("Not Yet Implemented");
            }
            actual.append("\n\n");
        }
        Approvals.verify(actual.toString());
    }

    @Test
    public void testContractorTradesModel_Operator() throws Exception {
        permissions.setAccountType("Corporate");
        model = new ContractorTradesModel(permissions);
        Approvals.verify(getJoin());
    }

    @Test
    public void testContractorsModelWhereClause_ArchivedReportDeactivatedContractorsVisibleForOperator() throws Exception {
        permissions.setAccountType("Corporate");
        model = new ContractorsModel(permissions);
        List<Filter> filters = new ArrayList<>();
        Filter filter = new Filter();
        filter.setField(new Field("AccountStatus"));
        filter.setName("ACCOUNTSTATUS");
        filter.setOperator(QueryFilterOperator.In);
        filter.setValue("Deactivated");
        filters.add(filter);
        assertEquals("Account.status IN ('Active','Deactivated')", model.getWhereClause(filters));
    }

    @Test
    public void testContractorsModel_Admin() throws Exception {
        EntityFactory.addUserPermission(permissions, OpPerms.AllOperators);
        EntityFactory.addUserPermission(permissions, OpPerms.RiskRank);
        EntityFactory.addUserPermission(permissions, OpPerms.AllContractors);
        permissions.setAccountType("Admin");
        model = new ContractorsModel(permissions);
        Approvals.verify(getJoin());
    }

    @Test
    public void testContractorsModel_LicenseReport() throws Exception {
        EntityFactory.addUserPermission(permissions, OpPerms.ContractorLicenseReport);
        permissions.setAccountType("Admin");
        model = new ContractorsModel(permissions);
        Approvals.verify(getJoin());
    }

    @Test
    public void testContractorsModel_Billing() throws Exception {
        EntityFactory.addUserPermission(permissions, OpPerms.Billing);
        permissions.setAccountType("Admin");
        model = new ContractorsModel(permissions);
        String output = "Permissions: " + permissions + "\n" + getJoin();
        Approvals.verify(output);
    }

    @Test
    public void testContractorsModel_Corporate() throws Exception {
        permissions.setAccountType("Corporate");
        model = new ContractorsModel(permissions);
        Approvals.verify(getJoin());
    }

    @Test
    public void testContractorAuditOperatorsModel() throws Exception {
        EntityFactory.addUserPermission(permissions, OpPerms.AllContractors);
        model = new ContractorAuditOperatorsModel(permissions);
        Approvals.verify(getJoin());
    }

    @Test
    public void testContractorFeesModel_Billing() throws Exception {
        EntityFactory.addUserPermission(permissions, OpPerms.Billing);
        permissions.setAccountType("Admin");
        model = new ContractorFeesModel(permissions);
        Approvals.verify(getJoin());
    }

    @Test
    public void testInvoiceModel() throws Exception {
        EntityFactory.addUserPermission(permissions, OpPerms.AllContractors);
        model = new InvoicesModel(permissions);
        Approvals.verify(getJoin());
    }

    @Test
    public void testInvoiceModel_UrlOperator() throws Exception {
        model = new InvoicesModel(permissions);
        Map<String, Field> fields = model.getAvailableFields();
        Field accountName = fields.get("AccountName".toUpperCase());
        assertTrue(accountName.getUrl().startsWith("ContractorView"));
    }

    @Test
    public void testInvoiceModel_UrlAdmin() throws Exception {
        EntityFactory.addUserPermission(permissions, OpPerms.AllContractors);
        model = new InvoicesModel(permissions);
        Map<String, Field> fields = model.getAvailableFields();
        Field accountName = fields.get("AccountName".toUpperCase());
        assertTrue(accountName.getUrl().startsWith("BillingDetail"));
    }

    @Test
    public void testPaymentModel() throws Exception {
        EntityFactory.addUserPermission(permissions, OpPerms.AllContractors);
        model = new PaymentsModel(permissions);
        Approvals.verify(getJoin());
    }

    @Test
    public void testGetWhereClause_COModel() throws Exception {
        EntityFactory.addUserPermission(permissions, OpPerms.AllContractors);
        model = new ContractorOperatorsModel(permissions);
        assertEquals("", model.getWhereClause(new ArrayList<Filter>()));
    }

    @Test
    public void testGetWhereClause_COModel_asOperator() throws Exception {
        permissions.setAccountType("Operator");
        model = new ContractorOperatorsModel(permissions);
        assertEquals("Account.status IN ('Active') AND ContractorOperator.opID IN (1100) AND ContractorOperator.workStatus = 'Y'", model.getWhereClause(new ArrayList<Filter>()));
    }

    @Test
    public void testGetWhereClause_COModel_asCorporate() throws Exception {
        permissions.setAccountType("Corporate");
        model = new ContractorOperatorsModel(permissions);
        assertEquals("Account.status IN ('Active') AND ContractorOperator.opID IN ()", model.getWhereClause(new ArrayList<Filter>()));
    }

    @Test
    public void testPOCModel() throws Exception {
        EntityFactory.addUserPermission(permissions, OpPerms.SalesCommission);
        model = new PaymentOperatorCommissionsModel(permissions);
        Approvals.verify(getJoin());
    }

    private String getJoin() {
        List<String> fieldValues = new ArrayList<String>();

        for (Field field : model.getAvailableFields().values()) {
            fieldValues.add(field.getName());
        }
        Collections.sort(fieldValues);
        return Joiner.on("\n").join(fieldValues);
    }

}
