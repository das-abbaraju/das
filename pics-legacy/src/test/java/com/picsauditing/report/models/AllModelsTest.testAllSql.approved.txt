Model: Accounts
Permission: Permissions{permissions=[]}
----------------------------------------
SELECT *
FROM accounts AS Account

Model: AuditTypes
Permission: Permissions{permissions=[]}
----------------------------------------
SELECT *
FROM audit_type AS AuditType

Model: ContractorAudits
Permission: Permissions{permissions=[]}
----------------------------------------
SELECT *
FROM contractor_audit AS Audit
JOIN contractor_info AS Contractor ON Audit.conID = Contractor.id
JOIN accounts AS Account ON Contractor.id = Account.id AND Account.type = 'Contractor'
JOIN audit_type AS AuditType ON Audit.auditTypeID = AuditType.id

Model: ContractorAuditDatas
Permission: Permissions{permissions=[]}
----------------------------------------
SELECT *
FROM pqfData AS AuditData
JOIN contractor_audit AS Audit ON AuditData.auditID = Audit.id
JOIN audit_type AS AuditType ON Audit.auditTypeID = AuditType.id
JOIN contractor_info AS Contractor ON Audit.conID = Contractor.id
JOIN accounts AS Account ON Contractor.id = Account.id AND Account.type = 'Contractor'
JOIN audit_question AS Question ON AuditData.questionID = Question.id
WHERE (1=0) 

Model: ContractorAuditFiles
Permission: Permissions{permissions=[]}
----------------------------------------
SELECT *
FROM contractor_audit_file AS ContractorAuditFile
JOIN contractor_audit AS Audit ON ContractorAuditFile.auditID = Audit.id
JOIN contractor_info AS ContractorInfo ON Audit.conID = ContractorInfo.id
JOIN accounts AS Account ON ContractorInfo.id = Account.id AND Account.type = 'Contractor'
WHERE (1=0 AND AuditOperator.visible = 1) 

Model: ContractorAuditOperators
Permission: Permissions{permissions=[]}
----------------------------------------
SELECT *
FROM contractor_audit_operator AS AuditOperator
JOIN accounts AS AuditOperatorAccount ON AuditOperator.opID = AuditOperatorAccount.id
JOIN contractor_audit AS Audit ON AuditOperator.auditID = Audit.id
JOIN audit_type AS AuditType ON Audit.auditTypeID = AuditType.id
JOIN contractor_info AS Contractor ON Audit.conID = Contractor.id
JOIN accounts AS Account ON Contractor.id = Account.id AND Account.type = 'Contractor'
WHERE (1=0 AND AuditOperator.visible = 1) 

Model: ContractorFees
Permission: Permissions{permissions=[]}
----------------------------------------
SELECT *
FROM contractor_fee AS ContractorFee
JOIN contractor_info AS Contractor ON ContractorFee.conID = Contractor.id
JOIN accounts AS Account ON Contractor.id = Account.id AND Account.type = 'Contractor'
JOIN invoice_fee AS Fee ON ContractorFee.newLevel = Fee.id
WHERE (1=0) 

Model: ContractorFlagDatas
Permission: Permissions{permissions=[]}
----------------------------------------
SELECT *
FROM flag_data AS FlagData
JOIN contractor_operator AS ContractorFlag ON FlagData.conID = ContractorFlag.conID AND FlagData.opID = ContractorFlag.opID
JOIN accounts AS ContractorOperatorOperator ON ContractorFlag.opID = ContractorOperatorOperator.id
JOIN flag_criteria_operator AS OperatorCriteria ON FlagData.criteriaID = OperatorCriteria.criteriaID AND FlagData.opID = OperatorCriteria.opID
JOIN flag_criteria AS FlagCriteria ON FlagData.criteriaID = FlagCriteria.id
JOIN accounts AS Account ON FlagData.conID = Account.id
WHERE (1=0) 

Model: ContractorNumbers
Permission: Permissions{permissions=[]}
----------------------------------------
SELECT *
FROM contractor_operator_number AS OperatorContractorNumber
JOIN contractor_info AS Contractor ON OperatorContractorNumber.conID = Contractor.id
JOIN accounts AS Account ON Contractor.id = Account.id AND Account.type = 'Contractor'
WHERE (1=0) 

Model: ContractorOperators
Permission: Permissions{permissions=[]}
----------------------------------------
SELECT *
FROM contractor_operator AS ContractorOperator
JOIN accounts AS ContractorOperatorOperator ON ContractorOperator.opID = ContractorOperatorOperator.id
JOIN contractor_info AS Contractor ON ContractorOperator.conID = Contractor.id
JOIN accounts AS Account ON Contractor.id = Account.id AND Account.type = 'Contractor'
WHERE (1 = 0) 

Model: Contractors
Permission: Permissions{permissions=[]}
----------------------------------------
SELECT *
FROM contractor_info AS Contractor
JOIN accounts AS Account ON Contractor.id = Account.id AND Account.type = 'Contractor'
WHERE (1=0) 

Model: ContractorSearch
Permission: Permissions{permissions=[]}
----------------------------------------
SELECT *
FROM contractor_info AS Contractor
JOIN accounts AS Account ON Contractor.id = Account.id AND Account.type = 'Contractor'
WHERE (Account.status NOT IN ('Demo','Deleted')) 

Model: ContractorStatistics
Permission: Permissions{permissions=[]}
----------------------------------------
SELECT *
FROM contractor_info AS Contractor
JOIN accounts AS Account ON Contractor.id = Account.id AND Account.type = 'Contractor'
JOIN (SELECT ca1.conID, ca1.auditFor,  CASE WHEN pd1.questionID = 8812 THEN 'Fatalities' ELSE 'TRIR' END AS rateType, pd1.answer  FROM contractor_audit ca1	 JOIN pqfdata pd1 ON ca1.id = pd1.auditID  WHERE ca1.auditTypeID = 11  AND pd1.questionID IN (8812,8977)) AS ContractorStatistics ON Contractor.id = ContractorStatistics.conID
WHERE (1=0) 

Model: EmployeeCompetencies
Permission: Permissions{permissions=[]}
----------------------------------------
SELECT *
FROM employee_competency AS EmployeeCompetency
JOIN operator_competency AS EmployeeCompetencyCompetency ON EmployeeCompetency.competencyID = EmployeeCompetencyCompetency.id
JOIN employee AS EmployeeCompetencyEmployee ON EmployeeCompetency.employeeID = EmployeeCompetencyEmployee.id
JOIN accounts AS EmployeeCompetencyEmployeeAccount ON EmployeeCompetencyEmployee.accountID = EmployeeCompetencyEmployeeAccount.id

Model: ForcedFlags
Permission: Permissions{permissions=[]}
----------------------------------------
SELECT *
FROM (SELECT co.conID AS conID, co.opID AS opID, co.forceBegin, co.forceEnd, co.forcedBy, co.forceFlag, 'Overall' AS label FROM contractor_operator co WHERE co.forceFlag IS NOT NULL UNION SELECT fdo.conID, fdo.opID, fdo.updateDate AS forceBegin, fdo.forceEnd, fdo.updatedBy AS forcedBy, fdo.forceFlag, CONCAT(fc1.id, '.label') AS label FROM flag_data_override fdo JOIN flag_criteria fc1 ON fdo.criteriaID = fc1.id WHERE fdo.forceFlag IS NOT NULL ) AS ContractorFlag
JOIN contractor_info AS ContractorFlagContractor ON ContractorFlag.conID = ContractorFlagContractor.id
JOIN accounts AS Account ON ContractorFlagContractor.id = Account.id AND Account.type = 'Contractor'

Model: ForcedFlagPercents
Permission: Permissions{permissions=[]}
----------------------------------------
SELECT *
FROM (SELECT co.opID AS opID, co.conID as conID, co.workStatus, (ROUND(COUNT(DISTINCT CASE WHEN co.forceFlag IS NOT NULL OR fdo.forceFlag IS NOT NULL THEN co.`conID` ELSE NULL END)/COUNT(DISTINCT conID)*100,2)) AS percentForced
FROM contractor_operator co
JOIN accounts a ON co.conID = a.id AND a.status = 'Active'
LEFT JOIN flag_data_override fdo ON fdo.`conID` = co.`conID` AND fdo.`opID` = co.`opID`
GROUP BY co.opID) AS ForcedFlag
JOIN accounts AS ContractorOperatorOperator ON ForcedFlag.opID = ContractorOperatorOperator.id
WHERE (1 = 0) 

Model: InvoiceItems
Permission: Permissions{permissions=[]}
----------------------------------------
SELECT *
FROM invoice_item AS InvoiceItem
JOIN invoice_fee AS InvoiceItemFee ON InvoiceItem.feeID = InvoiceItemFee.id
JOIN invoice AS Invoice ON InvoiceItem.invoiceID = Invoice.id
JOIN accounts AS Account ON Invoice.accountID = Account.id
JOIN contractor_info AS Contractor ON Account.id = Contractor.id
WHERE (Invoice.tableType = 'I') 

Model: Invoices
Permission: Permissions{permissions=[]}
----------------------------------------
SELECT *
FROM invoice AS Invoice
JOIN accounts AS Account ON Invoice.accountID = Account.id
JOIN contractor_info AS Contractor ON Account.id = Contractor.id
WHERE (Invoice.tableType = 'I') 

Model: Operators
Permission: Permissions{permissions=[]}
----------------------------------------
SELECT *
FROM accounts AS Account
WHERE (1=0) 

Model: OperatorAccountUsers
Permission: Permissions{permissions=[]}
----------------------------------------
SELECT *
FROM account_user AS AccountUser
JOIN accounts AS Account ON AccountUser.accountID = Account.id

Model: OperatorUsers
Permission: Permissions{permissions=[]}
----------------------------------------
SELECT *
FROM users AS User
JOIN accounts AS Account ON User.accountID = Account.id

Model: PaymentCommissions
Permission: Permissions{permissions=[]}
----------------------------------------
SELECT *
FROM payment_commission AS PaymentCommission
JOIN invoice_commission AS InvoiceCommission ON PaymentCommission.commissionID = InvoiceCommission.id
JOIN invoice AS Invoice ON InvoiceCommission.invoiceID = Invoice.id
JOIN accounts AS Account ON Invoice.accountID = Account.id
JOIN account_user AS AccountUser ON InvoiceCommission.accountUserID = AccountUser.id

Model: Reports
Permission: Permissions{permissions=[]}
----------------------------------------
SELECT *
FROM report AS Report
WHERE (Report.deleted = 0) 

Model: ReportColumns
Permission: Permissions{permissions=[]}
----------------------------------------
SELECT *
FROM report_column AS ReportColumn
JOIN report AS Report ON ReportColumn.reportID = Report.id

Model: ReportFilters
Permission: Permissions{permissions=[]}
----------------------------------------
SELECT *
FROM report_filter AS ReportFilter
JOIN report AS Report ON ReportFilter.reportID = Report.id

Model: ReportPermissions
Permission: Permissions{permissions=[]}
----------------------------------------
SELECT *
FROM report AS Report

Model: ReportUsers
Permission: Permissions{permissions=[]}
----------------------------------------
SELECT *
FROM report_user AS ReportUser
JOIN report AS Report ON ReportUser.reportID = Report.id
JOIN users AS ReportUserUser ON ReportUser.userID = ReportUserUser.id

Model: UserAssignments
Permission: Permissions{permissions=[]}
----------------------------------------
SELECT *
FROM user_assignment AS UserAssignment

Model: Users
Permission: Permissions{permissions=[]}
----------------------------------------
SELECT *
FROM users AS User
JOIN accounts AS Account ON User.accountID = Account.id

